python/lsst/sims/movingObjects/chebyValues.py

```diff
@@ -135,21 +145,37 @@ def getEphemerides(self, time, objIds=None):
                 objIds = np.array([objIds])
             objMatch = np.in1d(self.coeffs['objId'], objIds)
             ephemerides['objId'] = objIds
-        ephemerides['time'] = np.zeros(len(ephemerides['objId']), float) + time
+        # Now find ephemeris values.
+        ephemerides['time'] = np.zeros((len(ephemerides['objId']), ntimes), float) + time
         for k in self.ephemerisKeys:
-            ephemerides[k] = np.zeros(len(ephemerides['objId']), float)
-        segments = np.where((self.coeffs['tStart'][objMatch] <= time) &
-                            (self.coeffs['tEnd'][objMatch] > time))[0]
-        for i, segmentIdx in enumerate(segments):
-            ephemeris = self._evalSegment(segmentIdx, time, objMatch)
-            for k in self.ephemerisKeys:
-                ephemerides[k][i] = ephemeris[k]
-            ephemerides['objId'][i] = self.coeffs['objId'][objMatch][segmentIdx]
+            ephemerides[k] = np.zeros((len(ephemerides['objId']), ntimes), float)
+        for it, t in enumerate(time):
+            # Find subset of segments which contain the appropriate time.
+            # Look for simplest subset first.
+            segments = np.where((self.coeffs['tStart'][objMatch] <= t) &
+                                (self.coeffs['tEnd'][objMatch] > t))[0]
+            if len(segments) == 0:
+                segStart = self.coeffs['tStart'][objMatch].min()
+                segEnd = self.coeffs['tEnd'][objMatch].max()
+                if (segStart > t or segEnd < t):
+                    if not extrapolate:
+                        for k in self.ephemerisKeys:
+                            ephemerides[k][:,it] = np.nan
+                    else:
+                        # Find the segments to use to extrapolate the times.
+                        if segStart > t:
```

Comment:
Would it be more efficient to find the index of segStart and segEnd in lines 158 and 159 above by using `argmin()` and `argmax()`, instead of finding `min()` and `max()` and then using `np.where()` to get the index later?

I acknowledge that I am being nitpicky here...